### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by pauloue**
The previous `group_to_item` resulted in the invalid recipe rose + wool --> red wool being shown. It also didn't break out of the loop after finding a match.

`get_item_usages` now returns recipes with matching groups which means tools with a `group:stick` handle will be shown in the usages for stick.

Plus some other fixes, see the commits.

### CODE REVIEW

**Comment by kilbith** [comment]
> the invalid recipe rose + wool --> red wool being shown

Current master:
screenshot_20190107_143947

**Comment by pauloue** [comment]
I am seeing flowers in the recipes for red, white, and yellow wool. Probably a difference in the order of `minetest.registered_items`. The current code is wrong because for groupstrings like `group:dye,color_red` only the last group is checked.

File: init.lua [hunk 1]
```diff
@@ -223,50 +247,53 @@ local function get_recipe_fs(iX, iY, xoffset, recipe_num, recipes, show_usage)
 	local rows = ceil(maxn(items) / width)
 	local rightest, s_btn_size = 0
 
-	if recipe_type ~= "cooking" and (width > GRID_LIMIT or rows > GRID_LIMIT) then
+	if width > GRID_LIMIT or rows > GRID_LIMIT then
```
[the remark this answers is not in the document]
Comment (kilbith) [reply init.lua]: Ah yes, nevermind then.

File: init.lua [hunk 2]
```diff
@@ -88,6 +77,29 @@ craftguide.register_craft({
 	items  = {"default:stone"},
 })
 
+local function get_recipe(output)
+	local recipe = mt.get_craft_recipe(output)
+	if recipe.items then
+		return recipe
+	end
+	for _, custom_craft in ipairs(craftguide.custom_crafts) do
```
[the remark this answers is not in the document]
Comment (pauloue) [reply init.lua]: I think that's a pointless optimization. In a test I get a difference of 1 millisecond over a million calls of ipairs: https://gist.github.com/pauloue/c374a4cf481fdb6d7c60eebd5e65247f. Here it's called once.

File: init.lua [hunk 3]
```diff
@@ -135,23 +147,35 @@ local function item_in_inv(inv, item)
...
-				end
-			end
+local function extract_groups(str)
+	return str:sub(7):split(",")
+end
+
+local function item_has_groups(item_groups, groups)
+	for _, group in ipairs(groups) do
+		if not item_groups[group] then
+			return false
```
[the remark this answers is not in the document]
Comment (pauloue) [reply init.lua]: The false doesn't get saved in memory and it's much clearer than returning an implicit nil. We're not simply exiting the function, false is the actual return value.

**Comment by kilbith** [comment]
Merged with some modifications, thanks.

https://github.com/minetest-mods/craftguide/commit/7f6dd430ce59ef749324413ffc908dec43b7331e

**Pull request closed without merging**